mongo-statefulset/mongos/template.yaml

```diff
@@ -0,0 +1,195 @@
+    strategy:
+      activeDeadlineSeconds: 21600
+      resources: {}
+      rollingParams:
+        intervalSeconds: 1
+        maxSurge: 25%
+        maxUnavailable: 25%
+        timeoutSeconds: 600
+        updatePeriodSeconds: 1
+      type: Rolling
+    template:
+      metadata:
+        labels:
+          app: mongodb
+          deploymentconfig: mongos
+          component: mongos
+      spec:
+        containers:
+        - name: mongos
+          resources:
+            limits:
+              memory: "${MEMORY_LIMIT}"
+          env:
+            - name: MONGODB_ADMIN_PASSWORD
+              value: ${MONGODB_ADMIN_PASSWORD}
+            - name: MONGODB_KEYFILE_VALUE
+              value: ${MONGODB_KEYFILE_VALUE}
+            - name: ENABLE_TLS
+              valueFrom:
+                configMapKeyRef:
+                  name: mongos-config
+                  key: ssl.enabled
+            - name: SSL_CA_PATH
+              valueFrom:
+                configMapKeyRef:
+                  name: mongos-config
+                  key: ssl.ca.path
+            - name: ADDITIONAL_SSL_OPTS
+              valueFrom:
+                configMapKeyRef:
+                  name: mongos-config
+                  key: additional.ssl.opts
+            - name: CONFIG_REPLSET_NAME
+              valueFrom:
+                configMapKeyRef:
+                  name: mongos-config
+                  key: config.rep.name
+            - name: CONFIG_REPLSET_SERVER
+              valueFrom:
+                configMapKeyRef:
+                  name: mongos-config
+                  key: config.server
+            - name: REPLSET_NAMES
+              valueFrom:
+                configMapKeyRef:
+                  name: mongos-config
+                  key: rep.names
+            - name: REPLSET_SERVERS
+              valueFrom:
+                configMapKeyRef:
+                  name: mongos-config
+                  key: rep.servers
+            - name: ADDITIONAL_STARTUP_OPTS
+              valueFrom:
+                configMapKeyRef:
+                  name: mongos-config
+                  key: additional.startup.opts
+          args:
+          - run-mongos
+          image: "${MONGODB_IMAGE}"
+          imagePullPolicy: IfNotPresent
+          ports:
+          - containerPort: 27017
+            protocol: TCP
+            failureThreshold: 3
+            periodSeconds: 10
+            successThreshold: 1
+            timeoutSeconds: 1
+          livenessProbe:
+            initialDelaySeconds: 60
+            tcpSocket:
+              port: 27017
+          readinessProbe:
+            initialDelaySeconds: 10
+            exec:
+              command:
+                - "mongo-is-ready"
+          resources: {}
+          terminationMessagePath: /dev/termination-log
+        dnsPolicy: ClusterFirst
+        restartPolicy: Always
+        securityContext: {}
+        terminationGracePeriodSeconds: 30
+- apiVersion: v1
+  kind: Service
```

Comment:
Originally this work was done in the context of AWS, so the template may potentially create some unused objects like this one. I had also left the storage class checked in too. Don't know if we mind or not.